### PR TITLE
Merge pull request #1 from mellanox-hpc/master

### PR DESCRIPTION
**Pull request by mosesr76**
syn-libibtest

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libverbs-tests-SimX-pr/34/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libverbs-tests-SimX-pr/35/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libverbs-tests-SimX-pr/36/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libverbs-tests-SimX-pr/37/ for details (Mellanox internal link).

File: tests/vxlan/smoke.cc [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+
+		int len = BUF_SIZ ;
+
+		INIT(ctx_recv.init());
+		INIT(ctx_send.init());
+		INIT(qp_send.init());
+		INIT(qp_recv.init());
+		INIT(mr_recv.init());
+		INIT(mr_send.init());
+		INIT(qp_recv.set_up_flow_rules(&flow_rules));
```
Comment (Artemy-Mellanox) [inline tests/vxlan/smoke.cc]: Better make add rule and recv part of test

File: tests/vxlan/smoke.cc [hunk 2]
```diff
@@ -0,0 +1,630 @@
...
+		attr.max_rd_atomic = 1;
+		flags = IBV_QP_STATE ;
+
+		DO(ibv_modify_qp(qp, &attr, flags));
+	}
+
+	void vxlan_create_rules( struct ibv_qp  *qp,
+                struct ibv_flow_attr *flow_rules) {
+
+                flow_create_result = ibv_create_flow(qp, flow_rules);
```
Comment (Artemy-Mellanox) [inline tests/vxlan/smoke.cc]: Here you can use SET() macro

File: tests/vxlan/smoke.cc [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+	}
+
+	void vxlan_destroy_rules() {
+
+		if (ibv_destroy_flow(flow_create_result)) {
```
Comment (Artemy-Mellanox) [inline tests/vxlan/smoke.cc]: Here you can use DO() macro

File: tests/vxlan/smoke.cc [hunk 4]
```diff
@@ -0,0 +1,630 @@
...
+
+struct ibvt_raw_qp : public ibvt_qp {
+	 ibvt_raw_qp(ibvt_env &e, ibvt_pd &p, ibvt_cq &c) : ibvt_qp(e, p, c) {}
+
+	uintptr_t start;
+	size_t length;
+        struct ibv_flow	*flow_create_result;
+
+
+ 	virtual void recv(ibv_sge sge) {
```
Comment (Artemy-Mellanox) [inline tests/vxlan/smoke.cc]: Why do you need to overwrite recv and post_send?

File: tests/vxlan/smoke.cc (hunk 2 above)
Comment (mosesr76) [inline tests/vxlan/smoke.cc]: done.

File: tests/vxlan/smoke.cc (hunk 3 above)
Comment (mosesr76) [inline tests/vxlan/smoke.cc]: done.

File: tests/vxlan/smoke.cc (hunk 2 above)
Comment (mosesr76) [inline tests/vxlan/smoke.cc]: done.

File: tests/vxlan/smoke.cc (hunk 4 above)
Comment (Artemy-Mellanox) [inline tests/vxlan/smoke.cc]: I just don't see you check it, i.e., where do you read  wr_id?

Comment (mosesr76) [inline tests/vxlan/smoke.cc]: done.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libverbs-tests-SimX-pr/39/ for details (Mellanox internal link).

**Pull request merged**